extensions/arc/src/test/ui/dashboards/postgresConnectionStrings.test.ts

```diff
@@ -0,0 +1,160 @@
+					}
+				}
+			},
+			service: {
+				type: '',
+				port: 0
+			},
+			storage: {
+				data: {
+					className: '',
+					size: ''
+				},
+				logs: {
+					className: '',
+					size: ''
+				},
+				backups: {
+					className: '',
+					size: ''
+				}
+			}
+		},
+		status: {
+			externalEndpoint: '127.0.0.1:5432',
+			readyPods: '',
+			state: '',
+			logSearchDashboard: '',
+			metricsDashboard: '',
+			podsStatus: [{
+				conditions: [{
+					lastTransitionTime: '',
+					message: '',
+					reason: '',
+					status: '',
+					type: '',
+				}],
+				name: '',
+				role: '',
+			}]
+		}
+	}
+};
+
+
+
+describe('postgresConnectionStringsPage', function (): void {
+	let controllerModel: ControllerModel;
+	let postgresModel: PostgresModel;
+	let azdataApi: azdataExt.IAzdataApi;
+	let postgresConnectionStrings: PostgresConnectionStringsPage;
+
+	afterEach(function (): void {
+		sinon.restore();
+	});
+
+	beforeEach(async () => {
+		// Stub the azdata CLI API
+		azdataApi = new FakeAzdataApi();
+		const azdataExt = TypeMoq.Mock.ofType<azdataExt.IExtension>();
+		azdataExt.setup(x => x.azdata).returns(() => azdataApi);
+		sinon.stub(vscode.extensions, 'getExtension').returns(<any>{ exports: azdataExt.object });
+
+		// Setup Controller Model
+		controllerModel = new FakeControllerModel();
+
+		//Stub calling azdata login and acquiring session
+		sinon.stub(controllerModel, 'acquireAzdataSession').returns(Promise.resolve(vscode.Disposable.from()));
+
+		// Setup PostgresModel
+		const postgresResource: PGResourceInfo = { name: 'pgt', resourceType: '' };
+		const registration: Registration = { instanceName: '', state: '', instanceType: ResourceType.postgresInstances };
+		postgresModel = new PostgresModel(controllerModel, postgresResource, registration, new AzureArcTreeDataProvider(TypeMoq.Mock.ofType<vscode.ExtensionContext>().object));
+
+		// Setp stub of show call
+		const postgresShow = sinon.stub().returns(FakePostgresServerShowOutput);
+		sinon.stub(azdataApi, 'arc').get(() => {
+			return { postgres: { server: { show(name: string) { return postgresShow(name); } } } };
+		});
+
+		// Setup the PostgresConnectionsStringsPage -- TODO modelviewmock get file createContainerBuilderMock
+		let { modelViewMock } = createModelViewMock();
+		postgresConnectionStrings = new PostgresConnectionStringsPage(modelViewMock.object, undefined!, postgresModel);
+	});
+
+	describe('getConnectionStrings', function (): void {
+
+		it('Strings container should be empty since postgres model has not been refreshed', async function (): Promise<void> {
+			should(postgresConnectionStrings['getConnectionStrings']()).be.empty();
+		});
+
+		it('String contain correct ip and port', async function (): Promise<void> {
+			// Call to provide external endpoint
+			await postgresModel.refresh();
+
+			postgresConnectionStrings['getConnectionStrings']().forEach(k => {
+				should(k.value.includes('127.0.0.1')).be.True();
+				should(k.value.includes('5432')).be.True();
```

Comment:
It'd be nice to get these values from the actual object so they aren't defined twice.  Something like:

```
endpoint = FakePostgresServerShowOutput.result.status.externalEndpoint.split(":")
... endpoint[0]
... endpoint[1]
```